hyperopt/fmin.py

```diff
@@ -197,9 +201,9 @@ def get_queue_len():
             with tqdm(total=N+qlen, file=orig_stdout, postfix='best loss: ?',
                       disable=not self.show_progressbar, dynamic_ncols=True,
                       ) as pbar:
-                while n_queued < N:
+                while n_queued < N and (timeit.default_timer() - self.start_time) < self.max_time:
                     qlen = get_queue_len()
-                    while qlen < self.max_queue_len and n_queued < N:
+                    while qlen < self.max_queue_len and n_queued < N and (timeit.default_timer() - self.start_time) < self.max_time:
```

Comment:
are u sure u need to check the time twice? can line 205 take long to run?

Reply:
It is not required, condition was optimized